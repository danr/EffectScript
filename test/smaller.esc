
effect iter<a> {
  yield(a): string
}

function walk(i: r ! iter<int> ! e): int ! e {
  handle i() {
    case yield(x), k:
      "\nbegin " + show(x) +
      k(" first " + show(x)) +
      "\nmiddle " + show(x) +
      k(" second " + show(x)) +
      "\nend " + show(x)
    case done(r): "\ndone:" + r
  }
}

function onetwo() {
  yield(1) + yield(2) + " onetwo done"
}

function main() {
  walk(onetwo)
}


