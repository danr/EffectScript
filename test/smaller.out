
begin 1
begin 2
done: first 1 first 2 onetwo done
middle 2
done: first 1 second 2 onetwo done
end 2
middle 1
begin 2
done: second 1 first 2 onetwo done
middle 2
done: second 1 second 2 onetwo done
end 2
end 1
